### ISSUE TITLE
Intel "enter" assembly not supported?

### ISSUE THREAD

**Issue by Owlz** [angr/simuvex#53]
Folks,

Messing around w/ a CTF chal and noticed that the "enter" routine for this binary wasn't correctly handled in this case. This is a 32-bit Windows PE file that I've manually set up state and started in the middle.

Here's an example:

```asm
sub_403B04 proc near

arg_0= dword ptr  8
arg_4= dword ptr  0Ch
arg_8= dword ptr  10h

enter   0, 0
push    [ebp+arg_8]
call    sub_4011CF
push    [ebp+arg_0]
push    [ebp+arg_4]
call    sub_401000
call    sub_40126F
leave
retn    0Ch
sub_403B04 endp
```

That "enter 0,0" was throwing off angr. The following hook appears to correctly update the behavior:

```python
def enter(s):
    s.stack_push(s.regs.ebp)
    s.regs.ebp = s.regs.esp
```

Maybe this is something for pyvex? Dunno. Regardless, it seems like it should be a fairly easy update.

**Comment by rhelmot** [comment]
Valgrind and vex don't support enter. https://bugs.kde.org/show_bug.cgi?id=88116

**Comment by rhelmot** [comment]
I agree entirely that this is a very silly thing to not support! It should be a simple patch to vex; Ill probably do it soon.
